src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -3968,6 +3968,17 @@ int ViewProviderSketch::defaultFontSizePixels() const
     return static_cast<int>(metrics.height());
 }
 
+qreal ViewProviderSketch::getDevicePixelRatio() const
+{
+    auto activeView = this->getActiveView();
+    if (activeView) {
+        auto glWidget = qobject_cast<Gui::View3DInventor*>(activeView)->getViewer()->getGLWidget();
```

Comment:
This could be rewritten like so:
```suggestion
    if (auto activeView = qobject_cast<Gui::View3DInventor*>(this->getActiveView())) {
        auto glWidget = activeView->getViewer()->getGLWidget();
```

This ensures that it will not segfault if active view for some reason is not what we expect.

Reply:
@kadet1090 thanks, and good catch on that cast. 

Are there any rules about when you are/are not in the Qt domain? I guess I should just check what the file is using.

Reply:
That is very loosely defined, but rule of thumb is that we should avoid using Qt types outside of code behind of various dialogs, custom widgets etc. I would say that is class extends QObject using Qt types is fine, but if not it would be better to avoid them if not strictly required.